empirical_photometry.ipynb

```diff
@@ -0,0 +1,490 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import tynt\n",
+    "from synphot.spectrum import SpectralElement\n",
+    "\n",
+    "sdss_bands = ['u', 'g', 'r', 'i', 'z']\n",
+    "\n",
+    "f = tynt.FilterGenerator()\n",
+    "\n",
+    "bandpasses = {}\n",
+    "for band in sdss_bands:\n",
+    "    filt = f.download_true_transmittance('SLOAN/SDSS.' + band)\n",
+    "    bp = SpectralElement(Empirical1D,\n",
+    "                         points=filt.wavelength,\n",
+    "                         lookup_table=filt.transmittance)\n",
+    "    # for boxy bandpasses, tapering them with taper() is recommended\n",
+    "    bandpasses[band] = bp.taper()\n",
+    "\n",
+    "    plt.plot(bp.waveset, bp(bp.waveset), label=band)\n",
+    "plt.legend(loc='upper right')\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also construct a bandpass from a file using <tt>synphot</tt>'s <a href=\"https://synphot.readthedocs.io/en/latest/api/synphot.spectrum.SpectralElement.html#synphot.spectrum.SpectralElement.from_file\">SpectralElement.from_file()</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once we have all of the effects on the source spectrum set, we can combine them together by simply multiplying the SpectralElement objects:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model the observation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 106,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING: Source spectrum is tapered. [synphot.observation]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from synphot.observation import Observation\n",
+    "\n",
+    "mags_extrap = {}\n",
+    "for band in sdss_bands:\n",
+    "    observation = Observation(spectrum, bandpasses[band], force='taper')\n",
+    "    mags_extrap[band] = observation.effstim('ABmag').value\n",
+    "\n",
+    "# SDSS mags aren't exactly equal to AB mags. Apply corrections\n",
```

Comment:
Could you provide a link explaining what these corrections are? I'm not familiar with them, and this might be important for explaining the discrepancy between the observed and predicted mags.

Reply:
Since we decided to ditch magnitudes it's kind of a moot point now, but if you're curious you can read about the corrections [here](https://www.sdss.org/dr12/algorithms/fluxcal/#SDSStoAB)